tests/test_edgeql_data_migration.py

```diff
@@ -8785,6 +8785,108 @@ async def test_edgeql_migration_annotation_05(self):
 
         await self.migrate("")
 
+    async def test_edgeql_migration_reset_optional_01(self):
+        await self.migrate(r'''
+            abstract type Person {
+                required property name -> str;
+            }
+
+            type PC extending Person;
+        ''')
+
+        await self.migrate(r'''
+            abstract type Person {
+                property name -> str;
+            }
+
+            type PC extending Person;
+        ''')
+
+        await self.migrate(r'''
+            abstract type Person {
+                required property name -> str;
+            }
+
+            type PC extending Person;
+        ''', user_input=['""'])
+
+    async def test_edgeql_migration_reset_optional_02(self):
+        await self.migrate(r'''
+            abstract type Person {
+                required property name -> str;
+            }
+
+            type PC extending Person {
+                overloaded required property name -> str;
+            }
+        ''')
+
+        await self.migrate(r'''
+            abstract type Person {
+                property name -> str;
+            }
+
+            type PC extending Person {
+                overloaded property name -> str;
+            }
+        ''')
+
+    async def test_edgeql_migration_reset_optional_03(self):
+        await self.migrate(r'''
+            abstract type Person {
+                required property name -> str;
+            }
+
+            type PC extending Person {
+                overloaded required property name -> str;
+            }
+        ''')
+
+        await self.migrate(r'''
+            abstract type Person {
+                optional property name -> str;
+            }
+
+            type PC extending Person {
+                overloaded optional property name -> str;
+            }
+        ''')
+
+    @test.xfail('''
+        This fails because we try to set the parent as required while the
+        child is still optional.
+
+        For this to work, we need to process the *child* first,
+        but in order for the reverse case above to work we need to
+        process the *parent* first.
+
+        I don't know if there is any way to fix this sort of thing without
+        exposing this kind of semantic understanding to ordering.
+
+        Maybe suppressing some kinds of intermediate-state errors during
```

Comment:
We can do that, but only if the intermediately-broken-state exists _within_ a tree corresponding to a single DDL command.  We already do this for deletions.  Please open a ticket so we don't forget to look into this later.

Reply:
One possible solution would be to add DDL syntax, e.g. `SET REQUIRED RECURSIVE` or something.